server/controllers/finance/purchase.js

```diff
@@ -3,152 +3,213 @@ var db = require('./../../lib/db');
+*/
+function list (req, res, next) {
+  'use strict';
+  var sql;
 
   sql =
-    'INSERT INTO primary_cash ' +
-      '(project_id, type, date, deb_cred_uuid, deb_cred_type, currency_id, ' +
-      'cash_box_id, account_id, cost, description, origin_id, uuid, user_id) ' +
-    'VALUES ' +
-      '(?,?,?,?,?,?,?,?,?,?,?,?,?);';
+    'SELECT purchase.uuid, purchase.reference, purchase.cost, purchase.discount, purchase.purchase_date, purchase.paid, ' +
+    'creditor.text, employee.name, employee.prenom, user.first, user.last ' +
+    'FROM purchase ' +
+    'JOIN creditor ON creditor.uuid = purchase.creditor_uuid ' +
+    'JOIN employee ON employee.id = purchase.purchaser_id ' +
+    'JOIN user ON user.id = purchase.emitter_id; ';
+
+  if (req.query.complete === '1') {
+    sql =  
+      'SELECT purchase.uuid, purchase.reference, purchase.cost, purchase.discount, purchase.purchase_date, purchase.paid, ' +
+      'creditor.text, employee.name, employee.prenom, user.first, user.last, ' +
+      'purchase.creditor_uuid, purchase.timestamp, purchase.note, purchase.paid_uuid, purchase.confirmed, purchase.closed, ' +
+      'purchase.is_direct, purchase.is_donation, purchase.emitter_id, purchase.is_authorized, purchase.is_validate, ' +
+      'purchase.confirmed_by, purchase.is_integration, purchase.purchaser_id, purchase.receiver_id ' +
+      'FROM purchase ' +
+      'JOIN creditor ON creditor.uuid = purchase.creditor_uuid ' +
+      'JOIN employee ON employee.id = purchase.purchaser_id ' +
+      'JOIN user ON user.id = purchase.emitter_id; ';
+  }
+
+  db.exec(sql)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
 
-  params = values(data);
-  params.push(primaryCashId);
-  params.push(userId);
+function detail (req, res, next) {
+  'use strict';
 
-  return db.exec(sql, params);
+  var purchaseUuid = req.params.uuid,
+    purchaseRow, purchaseItems;
+
+  var sqlPurchase =
+    'SELECT purchase.uuid, purchase.reference, purchase.cost, purchase.discount, purchase.purchase_date, purchase.paid, ' +
+    'creditor.text, employee.name, employee.prenom, user.first, user.last ' +
+    'FROM purchase ' +
+    'JOIN creditor ON creditor.uuid = purchase.creditor_uuid ' +
+    'JOIN employee ON employee.id = purchase.purchaser_id ' +
+    'JOIN user ON user.id = purchase.emitter_id ' +
+    'WHERE purchase.uuid = ?';
+
+  var sqlPurchaseItem =
+    'SELECT purchase_item.purchase_uuid, purchase_item.uuid, purchase_item.quantity, purchase_item.unit_price, ' +
+    'purchase_item.total, inventory.text ' +
+    'FROM purchase_item ' +
+    'JOIN inventory ON inventory.uuid = purchase_item.inventory_uuid ' +
+    'WHERE purchase_item.purchase_uuid = ? ';  
+
+  db.exec(sqlPurchase, [purchaseUuid])
+  .then(function (row) {
+    purchaseRow = row;
+    return db.exec(sqlPurchaseItem, [purchaseUuid]);
+  })
+  .then(function (rows) {     
+    if (purchaseRow.length === 0) { 
```

Comment:
Use application status codes instead of this check.  You should also move this check to the previous `.then()` statement.  The corrected code should be:
```js
db.exec(sqlPurchase, [purchaseUuid])
.then(function (row) {
  if (row.length === 0) {
   throw req.codes.ERR_NOT_FOUND();
  }
  purchaseRow = row;
  return db.exec(sqlPurchaseItem, [purchaseUuid]);
})
.then(function (rows) {
  // send data back to the client.
});
```